File: Qspiders/counter.js
let txt = "ABcd@1$z"

// console.log(length(txt));
// console.log(txt.length);

let upperCounter = 0
let lowerCounter = 0
let digitCounter = 0
let specialCounter = 0

for (let i = 0; i < txt.length; i++) {
    const currentI = txt[i];

    if (currentI >= 'A' && currentI <= 'Z') {
        upperCounter++;
    }
    else if (currentI >= 'a' && currentI <= 'z') {
        lowerCounter++;
    }
    else if (currentI >= '0' && currentI <= '9') {
        digitCounter++;
    }
    else {
        specialCounter++;
    }
}

console.log("count of UpperCase:", upperCounter)
console.log("count of Lowercase:", lowerCounter)
console.log("count of Digits:", digitCounter)
console.log("count of Special Characters:", specialCounter)

File: Qspiders/counter.py
# we need to count the upper case , lower case, digit, special case 


txt = "ABcd@1$z"

upperCount,LowerCount,DigitCount,specialCount = 0,0,0,0

s = 0

while s <= (len(txt) - 1):
    char = txt[s]

    if 'A' <= char <= 'Z':
        upperCount += 1
    elif 'a' <= char <= 'z':
        LowerCount += 1
    elif '0' <= char <= '9':
        DigitCount += 1
    else:
        specialCount += 1

    s += 1

print("count of UpperCase:", upperCount)
print("count of Lowercase:", LowerCount)
print("count of Digits:", DigitCount)
print("count of Special Characters:", specialCount)


# assignment : 
# 1. find out the no of primitive and non - primitive
l = [1,3.4,"hi", {"hi", "bye"}, 3+5j]